contracts/modules/RelayerModule.sol

```diff
@@ -0,0 +1,372 @@
+    * @param _refundAddress The address refunded to prevents front-running.
+    */
+    function execute(
+        address _wallet,
+        address _module,
+        bytes calldata _data,
+        uint256 _nonce,
+        bytes calldata _signatures,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        address _refundToken,
+        address _refundAddress
+    )
+        external
+        returns (bool)
+    {
+        uint startGas = gasleft();
+        require(startGas >= _gasLimit, "RM: not enough gas provided");
+        require(verifyData(_wallet, _data), "RM: Target of _data != _wallet");
+        require(isModule(_wallet, _module), "RM: module not authorised");
+        StackExtension memory stack;
+        (stack.requiredSignatures, stack.ownerSignatureRequirement) = IModule(_module).getRequiredSignatures(_wallet, _data);
+        require(stack.requiredSignatures > 0 || stack.ownerSignatureRequirement == OwnerSignature.Anyone, "RM: Wrong signature requirement");
+        require(stack.requiredSignatures * 65 == _signatures.length, "RM: Wrong number of signatures");
+        stack.signHash = getSignHash(
+            address(this),
+            _module,
+            0,
+            _data,
+            _nonce,
+            _gasPrice,
+            _gasLimit,
+            _refundToken,
+            _refundAddress);
+        require(checkAndUpdateUniqueness(
+            _wallet,
+            _nonce,
+            stack.signHash,
+            stack.requiredSignatures,
+            stack.ownerSignatureRequirement), "RM: Duplicate request");
+        require(validateSignatures(_wallet, stack.signHash, _signatures, stack.ownerSignatureRequirement), "RM: Invalid signatures");
+        // solium-disable-next-line security/no-low-level-calls
+        (stack.success, stack.returnData) = _module.call(_data);
+        refund(
+            _wallet,
+            startGas,
+            _gasPrice,
+            _gasLimit,
+            _refundToken,
+            _refundAddress,
+            stack.requiredSignatures,
+            stack.ownerSignatureRequirement);
+        emit TransactionExecuted(_wallet, stack.success, stack.returnData, stack.signHash);
+        return stack.success;
+    }
+
+    /**
+    * @dev Gets the current nonce for a wallet.
+    * @param _wallet The target wallet.
+    */
+    function getNonce(address _wallet) external view returns (uint256 nonce) {
+        return relayer[_wallet].nonce;
+    }
+
+    /**
+    * @dev Implementation of the getRequiredSignatures from the IModule interface.
+    * The method should not be called and will always revert.
+    * @param _wallet The target wallet.
+    * @param _data The data of the relayed transaction.
+    * @return always reverts.
+    */
+    function getRequiredSignatures(address _wallet, bytes calldata _data) external virtual override view returns (uint256, OwnerSignature) {
+        revert("RM: disabled method");
```

Comment:
Could we move this to BaseModule (thereby making `revert` the default behaviour of `getRequiredSignature()`) instead of copying this in both RelayerModule and SimpleUpgrader?

Reply:
Done, at the expense of having to explicitly define the `getRequiredSignatures` method in the `TransferManager` since it inherited 2 implementations of that method from `BaseModule` (via `BaseTransfer`) and `OnlyOwnerModule` .